Sources/Linq2DynamoDb.DataContext/EntityWrapper.cs

```diff
@@ -14,6 +16,31 @@ internal class EntityWrapper : IEntityWrapper
         private readonly IEntityKeyGetter _keyGetter;
         private Document _doc, _newDoc;
 
+        private PropertyInfo _entityVersionNumberProperty;
+        private bool _hasResolvedEntityVersionNumberProperty;
+
+        /// <summary>
+        /// Gets PropertyInfo for the entity's property that has [DynamoDBVersion] 
+        /// attribute or returns null if there is none.
+        /// </summary>
+        private PropertyInfo EntityVersionNumberProperty {
+            get {
+                if (!_hasResolvedEntityVersionNumberProperty) {
+                    _entityVersionNumberProperty = Entity
+                        .GetType()
+                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                        .Where(property =>
+                            property
+                                .GetCustomAttributes(typeof(DynamoDBVersionAttribute), true)
+                                .SingleOrDefault() != null
+                        ).SingleOrDefault();
```

Comment:
also just realized there are 2 levels of validation you might want for an entity.

1. should not have more than one `[DynamoDbVersion]` attribute on a single property (the inner `SingleOrDefault`). This actually should be enforced by the [`AttributeUsage`](https://github.com/aws/aws-sdk-net/blob/3ea604ff2c90cebbb7e76cb8da89a75955c2400b/sdk/src/Services/DynamoDBv2/Custom/DataModel/Attributes.cs#L122) of the attribute itself:

```
[AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, Inherited = true, AllowMultiple = false)]
    public sealed class DynamoDBVersionAttribute : DynamoDBRenamableAttribute
    {
    }
```

so i don't think you need this inner check and can just use `Any`.

2. should not have more than one property decorated with `[DynamoDbVersion]`. this is the outer `SingleOrDefault` and we should throw a nice exception if it's violated.